hatch/app.py

```diff
@@ -152,20 +162,23 @@ async def release_file(
     )
 
 
-@app.post("/workspace/{workspace}/release/{release_id}/{name:path}")
+@app.post("/workspace/{workspace}/release/{release_id}")
 def release_file_upload(
-    workspace: str, release_id: str, name: str, token: AuthToken = Depends(validate)
+    workspace: str,
+    release_id: str,
+    release_file: schema.ReleaseFile,
+    token: AuthToken = Depends(validate),
 ):
     """Upload a file from a release to job-server."""
     if token.scope != "upload":
         raise HTTPException(403, "Forbidden")
 
+    name = release_file.name
     workspace_dir = validate_workspace(workspace)
     release_dir = validate_release(workspace_dir, release_id)
     path = release_dir / name
     if not path.exists():
         raise HTTPException(404, f"File {name} not found in release {release_id}")
-
```

Comment:
Could you leave this blank line in?  It's really helpful to break up blocks of code for some folks (me!) to help parse them 🙂